dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt

```diff
@@ -14,30 +14,43 @@ import org.springframework.web.bind.annotation.RestController
 /**
  * Implementation of the API for data exchange
  * @param dataStore implementation of the DataStoreInterface that defines how uploaded data is to be stored
- * @param dataConnector implementation of the DataConnectorInterface that defines how to connect to the data
- * source (e.g. skyminder)
  */
 
 @RestController
-class DataController(
+abstract class DataController<T>(
     @Autowired @Qualifier("DefaultStore") var dataStore: DataStoreInterface,
-    @Autowired var dataConnector: DataConnectorInterface
-) : DataAPI {
+    var objectMapper: ObjectMapper
+) : DataAPI<T> {
+    private val dataType = getClazz().toString().substringAfterLast(".")
+
+    /**
+     * Method to get the class of the abstract T
+     */
+    abstract fun getClazz(): Class<T>
```

Comment:
Ich habe einen Vorschlag wie man das hier noch etwas vereinfachen kann in diesem Branch gemacht:
DALA-16_taxonomy_endpoints_simplify_generic_endpoint

Reply:
Laut https://github.com/d-fine/Dataland/actions/runs/2004412309 laufen die E2E Tests - Backend funktioniert da also noch. Bitte dann so einbauen.

Reply:
Fixed

Reply:
ok